packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -331,7 +331,11 @@ export function indexHtmlMiddleware(
       if (fs.existsSync(filename)) {
         try {
           let html = await fsp.readFile(filename, 'utf-8')
-          html = await server.transformIndexHtml(url, html, req.originalUrl)
+          html = await server.transformIndexHtml(
+            decodeURI(url),
```

Comment:
Because the document passes `req.originalUrl` to `transformIndexHtml`, I think `transformIndexHtml` expects an encoded URL.
https://vitejs.dev/guide/ssr.html#setting-up-the-dev-server
https://github.com/vitejs/vite/blob/4cc671f90415474cb0a90e210fe2ec36aa64a452/playground/ssr/server.js#L46-L50

Also, changing this will change the `ctx.path` of `transformIndexHtml` hook to a decoded URL from an encoded URL.
https://vitejs.dev/guide/api-plugin.html#transformindexhtml

I feel decoding the URL in `devHtmlHook` is safer.